app/controllers/portal.js

```diff
@@ -188,6 +190,30 @@ export default Ember.Controller.extend({
         })
         this.send('updateDOM')
       }
+    },
+    /**
+     * Updates the data object which is used in the frost select
+     * @param {String} username - The username of the updated user
+     * @param {String} token - The updated token
+     */
+    updateData: function (username, token) {
+      let data = this.get('data').map((item) => {
+        if (item.label === username.toString()) {
+          item.value = token
+        }
+        return item
+      })
+      this.set('data', Ember.A(data))
+    },
+    /**
+     * Logs the user out by calling the logout route. This route is being
+     * listened to by Passport and will logout the user and end the session. Then it will
+     * return successfully and we will transition to the index page.
+     */
+    logout: function () {
+      Ember.$.get('/logout', () => {
+        this.transitionToRoute('index')
+      })
```

Comment:
After hitting logout and then navigating to an authorized URL, I think you found that the server was either (a) not invalidating the user's token when logging out, or (b) not checking the user's token properly when visiting a page that requires authentication. Please check this and add some tests to verify you don't have a regression.